Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +124,155 @@ public void Create_PropertyDeclaredInBaseClass ()
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result1 = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock1.Object);
+
+      var notNullValidators1 = ((IPropertyValidationRule)result1).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators1.Length, Is.EqualTo(1));
+      Assert.That(notNullValidators1[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result2 = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock2.Object);
+
+      var notNullValidators2 = ((IPropertyValidationRule)result2).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators2.Length, Is.EqualTo(2));
+      Assert.That(notNullValidators2[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+      Assert.That(notNullValidators2[1].ValidationMessage.ToString(), Is.EqualTo("newTest"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_IgnoresValidationMessagesForValidatorsWithoutADeferredInitializationValidationMessage ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var propertyValidators = ((IPropertyValidationRule)result).Validators;
+
+      Assert.That(propertyValidators, Is.Empty);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithValidationMessageFactoryReturnsNull_ThrowsInvalidOperationException ()
     {
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns((ValidationMessage)null);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+
+      Assert.That(
+          () => _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo("The IValidationMessageFactory did not return a result for NotNullValidator applied to property 'UserName' on "
+                                    + "type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
```

Comment:
I found a probable message bug in prod code: we try to use single-quotes for parameter values. => `NotNullValidator` should have quotes. Please update the test and prod code. The `IValidationMessageFactory` string is an edge case. here, the sentence / readability works better if it is considered the subject and part of the sentence instead of information provided to the user.

Reply:
So, I guess this message bug is also in `AddingObjectValidationRuleCollector1.cs` ?

Reply:
I didn't verify it but it sounds reasonable to me :)